csrc/python_frontend/fusion_definition.cpp

```diff
@@ -170,6 +175,26 @@ std::vector<at::Tensor> FusionDefinition::execute(
 
   auto scheds = fusionCache()->queryFusionSchedules(id().value());
 
+#ifdef NVFUSER_DISTRIBUTED
+  static std::unique_ptr<Communicator> comm = nullptr;
```

Comment:
```suggestion
  static Communicator* comm = new Communicator();
```

unique_ptr here doesn't provide benefits -- it'll be deleted when the program exits anyway. However, a static, non-trivially destructible object can potentially create order problems. 

https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

Reply:
touche.

Thanks for catching this one!